crates/oxc_isolated_declarations/src/lib.rs

```diff
@@ -405,6 +405,67 @@ impl<'a> IsolatedDeclarations<'a> {
     }
 
     pub fn report_error_for_expando_function(&self, stmts: &oxc_allocator::Vec<'a, Statement<'a>>) {
+        let mut assignable_properties_for_namespace = FxHashMap::<&str, FxHashSet<Atom>>::default();
+        for stmt in stmts {
+            let Statement::ExportNamedDeclaration(decl) = stmt else { continue };
+            let Some(Declaration::TSModuleDeclaration(decl)) = &decl.declaration else { continue };
```

Comment:
What about this case?
```ts
namespace NS { }
NS.a = 0;
export { NS }
```

Reply:
Do you mean the case where there is no function declaration? I believe this is already allowed and handled correctly.

Reply:
> Do you mean the case where there is no function declaration? I believe this is already allowed and handled correctly.

I mean namespace does not export. I saw you only handled `ExportNamedDeclartion`

Reply:
I see. This is consistent with TypeScript's behavior. If the namespace isn't exported, the error will still be reported.
[playground](https://www.typescriptlang.org/dev/bug-workbench/?target=7&noCheck=true&isolatedDeclarations=true&emitDeclarationOnly=true&ts=5.5.0-dev.20240502#code/PTAEAEGcBcCcEsDG0Bco4FcCmAoEEATLRAGwENYzp4B7AOzU133DpoGEALYga0dmx4w4LAFt40ACLFylavQDydEgE9+glvEg1y0LAWmkKVWnUjrmwsiRI0A7gClzoAGbXIliJE72AouOghLx87fwk9AgAxeBIsNDIAOgIE6EgcIPAXGKw6MlE40ETUnCwADwAHGlhoVww6ZFNXGhoACgBKNAA3GngCUABvAF88ACpQMsrq0BGwXPzIcrJELCaaAZxQcYqqmtiasjQ6DFEAIyxYAG4cYZdmhLJQAF5QABYAJiA)

Reply:
Cool, Can you add a case to cover it?

Reply:
Yes, done